### PR TITLE
Cmsis

### PR DESCRIPTION
**Pull request by psiddh**
### Summary
[PLEASE REMOVE] See [CONTRIBUTING.md's Pull Requests](https://github.com/pytorch/executorch/blob/main/CONTRIBUTING.md#pull-requests) for ExecuTorch PR guidelines.

[PLEASE REMOVE] If this PR closes an issue, please add a `Fixes #<issue-id>` line.

[PLEASE REMOVE] If this PR introduces a fix or feature that should be the upcoming release notes, please add a "Release notes: <area>" label. For a list of available release notes labels, check out [CONTRIBUTING.md's Pull Requests](https://github.com/pytorch/executorch/blob/main/CONTRIBUTING.md#pull-requests).

### Test plan
[PLEASE REMOVE] How did you test this PR? Please write down any manual commands you used and note down tests that you have written if applicable.

### CODE REVIEW

File: backends/cortex_m/ops/op_add.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      "out.scalar_type() %" PRId8 " is not float",
+      static_cast<int8_t>(out.scalar_type()));
+
+  // Check dtype is int8 (Char)
+  ET_CHECK_MSG(
+      dtype == ScalarType::Char,
+      "dtype %" PRId8 " is not int8 (Char)",
+      static_cast<int8_t>(dtype));
+  
+  assert(false);
```
Comment (digantdesai) [inline backends/cortex_m/ops/op_add.cpp:43]: this will be a no-op for a RELEASE CMake build.

File: backends/cortex_m/passes/replace_quant_nodes_pass.py [hunk 2]
```diff
@@ -31,6 +31,10 @@ def _is_qualified_int8_node(args) -> bool:
     def __init__(self):
         super().__init__()
         self.op_replacements = {
+            exir_ops.edge.aten.add.Tensor: {
```
Comment (digantdesai) [inline backends/cortex_m/passes/replace_quant_nodes_pass.py:38]: Did you print your graph before and after this pass to see if this replacement is actually happening?

File: runtime/core/named_data_map.h [hunk 3]
```diff
@@ -27,7 +27,7 @@ namespace ET_RUNTIME_NAMESPACE {
  * Interface to access and retrieve data via name.
  * See executorch/extension/flat_tensor/ for an example.
  */
-class ET_EXPERIMENTAL NamedDataMap {
+class NamedDataMap {
```
Comment (digantdesai) [inline runtime/core/named_data_map.h:30]: accidental?

Comment (psiddh) [reply runtime/core/named_data_map.h:30]: Will remove it.

File: backends/cortex_m/passes/replace_quant_nodes_pass.py (hunk 2 above)
Comment (psiddh) [reply backends/cortex_m/passes/replace_quant_nodes_pass.py:38]: Yes , for the softmax usecase, I see the following

[**ReplaceQuantNodesPass**] Operator called: <EdgeOpOverload: aten._softmax.default>: schema = aten::_softmax(Tensor self, int dim, bool half_to_float) -> Tensor, Args: (<executorch.exir.pass_base.ProxyValue object at 0x7fa982bc03a0>, 0, False)
[**ReplaceQuantNodesPass**] Replacing <EdgeOpOverload: aten._softmax.default>: schema = aten::_softmax(Tensor self, int dim, bool half_to_float) -> **_Tensor with cortex_m.softmax_**
